modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java

```diff
@@ -116,6 +124,68 @@ public long getObjectDefinitionId() {
 		return objectDefinition.getObjectDefinitionId();
 	}
 
+	public ArrayList<JSONObject> getObjectRelationshipDeletionTypes() {
```

Comment:
Hey Dan, it is not common to create `ArrayList<JSONObject>`. An array or list of JSONObject is a JSONArray. There are many ways to create a JSONArray, but we can use a Util class and do like this `JSONArray jsonArray = JSONFactoryUtil.createJSONArray();`. After this, we populate with JSONObject like this `jsonArray.put()`.

Looking at your code, we could replace the `deletionTypes.add() `with the `jsonArray.put()` and continue to use the `JSONUtil.put()`. Do you agree?

Reply:
After your refactoring, maybe it won't be needed a local variable. You could create and populate your JSONArray directly to the return statement.

Reply:
Thanks for the Review @marcelabc :smile: